### DISCUSSION TITLE
click action unuseable when back

### DISCUSSION THREAD

**Discussion by suchuyang** [flet-dev/flet#3706] [Q&A]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the bug

I write some views, there is a listview in the first page and i click a button on a cell to enter another view. 
Back it.
So, the click action i clicked before does not work.

Here is the whole code sample.
[backexample.zip](https://github.com/user-attachments/files/16357816/backexample.zip)

### Code sample

<details open><summary>main.py</summary>

```main.py
import flet as ft
from ClassListView import *
from practiceview import *

def main(page: ft.Page):
    page.title = "CG练习统计程序"
    page.vertical_alignment = ft.MainAxisAlignment.CENTER
    page.horizontal_alignment=ft.CrossAxisAlignment.CENTER

    def route_change(e):
        if page.route == "newpage":
            classview = ClassList(page.width,page.height-100)
            classview.route = page.route
            page.views.append(classview)
        if "classpageof_" in page.route:
            practiceview = PracticeListView("1","practice")
            page.views.append(practiceview)
        page.update()

    def view_pop(view):
        page.views.pop()
        page.update()
    page.on_route_change = route_change
    page.on_view_pop = view_pop 
    
    def click(para):
        page.go(para.control.url_target)


    page.add(ft.TextButton("newpage",on_click=click,url_target="newpage"))

ft.app(main)
```

</details>
<details open><summary>ClassListView .py</summary>

```ClassListView.py
import flet as ft

class ClassList(ft.ListView):
    def __init__(self,w,h):
        super().__init__()
        for i in range(3):
            cell = ft.Row([
                 ft.Text(f"{i} cell", size=20),
                 ft.IconButton(icon=ft.icons.ARROW_RIGHT,on_click=self.cellclick,url_target=f"classpageof_{i}")
            ])
            self.controls.append(cell)

    def cellclick(self,para:ft.ControlEvent):
        print("click class:")
        print("name=",para.name)
        print("data=",para.data)
        print("target=",para.control.url_target)
        print("page=",self.page)
        self.page.go(para.control.url_target)
```

</details>

</details>
<details open><summary>practiceview.py</summary>

```practiceview.py
import flet as ft

class PracticeListView(ft.View):
    def __init__(self,classid,classname):
        super().__init__()
        self.classid = classid
        self.classname = classname
        self.bgcolor = "#A3AC99"
        self.vertical_alignment=ft.MainAxisAlignment.CENTER
        self.horizontal_alignment=ft.CrossAxisAlignment.CENTER
        #来一个appbar
        self.appbar = ft.AppBar(
                        title=ft.Text(f"练习"),
                    )
        self.controls.append(self.appbar)

```

</details>

### To reproduce

1、Run the sample code and click the newpage
![image](https://github.com/user-attachments/assets/17af3d4b-5b27-4fb6-b65f-3220183a897d)

2、click a cell right button 
![image](https://github.com/user-attachments/assets/5f1019a5-9feb-4499-a6a6-a89e939e40d4)

3、at the appbar view，click back button
![image](https://github.com/user-attachments/assets/c57d40f1-7bd8-425e-99e6-4ecd81d9be6f)

4、click the cell button again
![image](https://github.com/user-attachments/assets/2a4b29e2-4793-4e93-981d-d2a4d975926a)

5、click another cell button ，it work
![image](https://github.com/user-attachments/assets/faed3583-7712-494e-b516-434ff2f16664)

### Screenshots / Videos

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/564d6c91-df90-466f-b189-152ba1f5b772

</details>

### Operating System

Windows

### Operating system details

win11

### Flet version

0.23.2

### Regression

Yes, it used to work in a previous Flet version (please specify the version in additional details)

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by ndonkoHenri** [comment]
Your popping callback should look like [this](https://flet.dev/docs/getting-started/navigation-and-routing#:~:text=def%20view_pop(view)%3A%0A%20%20%20%20%20%20%20%20page.views.pop()%0A%20%20%20%20%20%20%20%20top_view%20%3D%20page.views%5B%2D1%5D%0A%20%20%20%20%20%20%20%20page.go(top_view.route)).

**Reply by suchuyang** [reply]
This method will create a new view, so all data will be reload again. I want to use the view object create before, because some setting is still on the back page.
